clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -35,16 +36,17 @@ private object ApiClient {
 
   /** @return an ApiClient, reusing an existing one for this URL if possible.
    */
-  def get(
-      apiUrl: String,
-      accessKey: String,
-      secretKey: String,
-      connectionTimeoutSec: String = "",
-      readTimeoutSec: String = ""
-  ): ApiClient = clients.get(
-    ClientKey(apiUrl, accessKey),
+  def get(conf: APIConfigurations): ApiClient = clients.get(
+    ClientKey(conf.apiUrl, conf.accessKey),
     new Callable[ApiClient] {
-      def call() = new ApiClient(apiUrl, accessKey, secretKey, connectionTimeoutSec, readTimeoutSec)
+      def call() = new ApiClient(
+        APIConfigurations(conf.apiUrl,
+                          conf.accessKey,
+                          conf.secretKey,
+                          conf.connectionTimeoutSec,
+                          conf.readTimeoutSec
+                         )
+      )
```

Comment:
I guess I am asking if we can just do this:
```suggestion
      def call() = new ApiClient(conf)
```

Reply:
I've considered it and am trying to remember why it didn't work..! :) 
I'm going ahead with the pr without addressing this comment for time reasons, will open an issue to track this improvement.